contracts/strategies/fluid/FluidStrategy.sol

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {InitializableAbstractStrategy, Helpers, IStrategyVault} from "../InitializableAbstractStrategy.sol";
+import {IFluidToken} from "./interfaces/IFluidToken.sol";
+
+/// @title Fluid strategy for USDs protocol
+/// @author Sperax Foundation
+/// @notice A yield earning strategy for USDs protocol
+/// @notice Important contract addresses:
+///         https://github.com/Instadapp/fluid-contracts-public/blob/main/deployments/deployments.md#lendingfactory
+contract FluidStrategy is InitializableAbstractStrategy {
+    using SafeERC20 for IERC20;
+
+    mapping(address => uint256) public allocatedAmount; // tracks the allocated amount for an asset.
+
+    error NoRewardIncentive();
+    error LimitReached();
+
+    /// @notice Initializer function of the strategy to initialize the state variables of InitializableAbstractStrategy
+    function initialize(
+        address _vault,
+        uint16 _depositSlippage, // 200 = 2%
+        uint16 _withdrawSlippage // 200 = 2%
+    ) external initializer {
+        InitializableAbstractStrategy._initialize(_vault, _depositSlippage, _withdrawSlippage);
+    }
+
+    /// @notice Provide support for asset by passing its lpToken address.
+    ///      This method can only be called by the contract owner
+    /// @param _asset    Address of the asset
+    /// @param _lpToken   Address of the corresponding platform token
+    function setPTokenAddress(address _asset, address _lpToken) external onlyOwner {
+        _setPTokenAddress(_asset, _lpToken);
+    }
+
+    /// @dev Remove a supported asset by passing its index.
+    ///       This method can only be called by the system owner
+    ///  @param _assetIndex Index of the asset to be removed
+    function removePToken(uint256 _assetIndex) external onlyOwner {
+        address asset = _removePTokenAddress(_assetIndex);
+        if (allocatedAmount[asset] != 0) {
+            revert CollateralAllocated(asset);
+        }
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function deposit(address _asset, uint256 _amount) external override nonReentrant {
+        Helpers._isNonZeroAmt(_amount, "Must deposit something");
+        address lpToken = _getPTokenFor(_asset);
+
+        // Checking for maximum deposit amount.
+        if (_amount > IFluidToken(lpToken).maxDeposit(address(this))) {
+            revert LimitReached();
+        }
+
+        allocatedAmount[_asset] += _amount;
+
+        // Doing the deposit.
+        IERC20(_asset).safeTransferFrom(msg.sender, address(this), _amount);
+        IERC20(_asset).forceApprove(lpToken, _amount);
+        uint256 sharesMinted = IFluidToken(lpToken).deposit(_amount, address(this));
+
+        uint256 sharesValue = IFluidToken(lpToken).convertToAssets(sharesMinted);
```

Comment:
We can use the value directly, no need to store in memory.

Reply:
We need the variable twice, one in the if statement check and the other while passing in the error.